### PR TITLE
Use new Drain API

### PR DESCRIPTION
**Pull request by JoshLove-msft**

### CODE REVIEW

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs [hunk 1]
```diff
@@ -1358,13 +1358,10 @@ public override async Task CloseAsync(CancellationToken cancellationToken)
...
                 // Allow in-flight messages to drain so that they do not remain locked by the service after closing the link.
-                // This should be updated when the AMQP library adds deterministic support for draining via an async method to remove the
-                // somewhat arbitrary delay and to also work for prefetch.
 
-                if (!_isSessionReceiver && !link.Settings.AutoSendFlow && link.LinkCredit > 0)
+                if (!_isSessionReceiver && link.LinkCredit > 0)
                 {
-                    link.IssueCredit(link.LinkCredit, true, AmqpConstants.NullBinary);
-                    await Task.Delay(200, cancellationToken).ConfigureAwait(false);
+                    await link.DrainAsyc(cancellationToken).ConfigureAwait(false);
```
Comment (jsquire) [inline sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs:1364]: That seems much nicer!

**Review by jsquire — APPROVED** [review]

File: sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs (hunk 1 above)
Comment (JoshLove-msft) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs:1364]: Agreed!

Comment (danielmarbach) [reply sdk/servicebus/Azure.Messaging.ServiceBus/src/Amqp/AmqpReceiver.cs:1364]: Ver cool!

**Pull request merged**